source/app/util/environment-check.ts

```diff
@@ -132,3 +134,53 @@ export default async function environmentCheck (): Promise<void> {
 
   global.log.info('Environment check complete.')
 }
+
+export interface BooleanError {
+  value: boolean
+  err?: string
+}
+
+/**
+ * Check if system supports a Tray.
+ *
+ * Returns {value: true} on Windows and MacOS.
+ * Returns {value: true} on Linux with Gnome desktop if Gnome Extension
+ * 'KStatusNotifierItem/AppIndicator Support' is installed, otherwise false.
+ * Returns {value: true} on Linux with other desktops. e.g. KDE, XFCE, etc.
+ *
+ * @return {*}  {Promise<BooleanError>} If supported, returns {value: true}.
+ *              If not supported, returns {value: false, err: error_string}.
+ *              error_string details why the Tray is not supported. Throws an
+ *              exception if an error occurred while checking Tray support.
+ */
+export async function isTraySupported (): Promise<BooleanError> {
+  const isLinux = process.platform === 'linux'
+  if (isLinux) {
+    if (process.env.XDG_CURRENT_DESKTOP === 'GNOME') {
+      return await new Promise<BooleanError>((resolve, reject) => {
+        const shellProcess = spawn('gsettings', [ 'get', 'org.gnome.shell', 'enabled-extensions' ])
+        let out = ''
+
+        shellProcess.stdout.on('data', (data) => {
+          out = out.concat(data.toString())
+        })
+
+        shellProcess.on('close', (code, signal) => {
+          if (code !== 0) {
+            resolve({ value: false, err: trans('system.error.tray_not_supported') })
+          } else if (out.includes("'appindicatorsupport@rgcjonas.gmail.com'")) {
+            resolve({ value: true })
+          } else {
+            resolve({ value: false, err: trans('system.error.tray_not_supported') })
```

Comment:
Is it necessary to declare an interface to return an error message? As there is only one case of returning the false, we could directly through the error. What do you think?

Reply:
@ratherto I see what you mean. Regarding directly throw the error: I remember "exceptions are for exceptional circumstances" and tray supported = false is a common occurance. Do you agree?

Reply:
Yes, I agree. Returning an error message is a good idea if we need to call this function multiple times. I have approved this PR